src/resources/2022/CombatLoversLocket.ts

```diff
@@ -0,0 +1,73 @@
+import { cliExecute, getLocketMonsters, Monster, toMonster } from "kolmafia";
+import { have as haveItem } from "../../lib";
+import { get } from "../../property";
+import { $item } from "../../template-string";
+import { clamp } from "../../utils";
+
+// eslint-disable-next-line libram/verify-constants
+export const locket = $item`Combat Lover's Locket`;
+
+export function have(): boolean {
+  return haveItem(locket);
+}
+
+/**
+ * Filters the set of all unlocked locket monsters to only the ones available to be locketed right now.
+ * @returns An array consisting of all Monsters you can fight with your locket right now.
+ */
+export function availableLocketMonsters(): Monster[] {
+  if (reminiscesLeft() === 0) return [];
+  return Object.entries(getLocketMonsters())
+    .filter(([, unused]) => unused)
+    .map(([name]) => toMonster(name));
+}
+
+/**
+ * Parses getLocketMonsters and returns the collection of all Monsters as an Array.
+ * @returns An array consisting of all Monsters you can hypothetically fight, regardless of whether they've been fought today.
+ */
+export function unlockedLocketMonsters(): Monster[] {
+  return Object.entries(getLocketMonsters()).map(([name]) => toMonster(name));
+}
+
+/**
+ * Determines how many reminisces remain by parsing the _locketMonstersFought property.
+ * @returns The number of reminisces a player has available; 0 if they lack the Locket.
+ */
+export function reminiscesLeft(): number {
+  return have()
+    ? clamp(
+        3 -
+          get("_locketMonstersFought")
```

Comment:
Only minor but I'd make an unexported fn here that gets splits and filters the pref, since you're doing that in two places